Sources/QuranAudioKit/Preferences/AudioPreferences.swift

```diff
@@ -7,19 +7,52 @@
 
 import Combine
 import Foundation
+import OrderedCollections
 import Preferences
 
 @available(iOS 13.0, *)
 public class AudioPreferences {
     public static let shared = AudioPreferences()
+
     private init() {}
 
     private static let lastSelectedReciterId = PreferenceKey<Int>(key: "LastSelectedQariId", defaultValue: 41)
     private static let audioEndKey = PreferenceKey<Int>(key: "audioEndKey", defaultValue: AudioEnd.juz.rawValue)
+    private static let recentReciterIds = PreferenceKey<[Int]>(key: "recentRecitersIdsKey", defaultValue: [])
+    private static let recentReciterIdsTransfomer = PreferenceTransformer<[Int], OrderedSet<Int>>(
+        rawToValue: { OrderedSet($0) },
+        valueToRaw: { Array($0) }
+    )
+    private static let maxNumOfRecentReciters = 3
 
     @TransformedPreference(audioEndKey, transformer: .rawRepresentable(defaultValue: .juz))
     public var audioEnd: AudioEnd
 
     @Preference(lastSelectedReciterId)
     public var lastSelectedReciterId: Int
+
+    @TransformedPreference(recentReciterIds, transformer: recentReciterIdsTransfomer)
+    public var recentReciterIds: OrderedSet<Int>
+
+    public func recentReciters(_ allReciters: [Reciter]) -> [Reciter] {
```

Comment:
This code doesn't seem to belong to the Audio preference as it converts from ID to Reciter and `updateRecentRecitersList` seems to do the other way around. I suggest you either:
1) Move them to ReciterDataRetriever and probably rename that struct, or
2) Create RecentRecitersService that encapsulates this logic.